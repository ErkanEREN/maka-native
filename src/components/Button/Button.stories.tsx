// src/components/Button/Button.stories.tsx

import React from "react";
import { Text } from "react-native";
import { storiesOf } from "@storybook/react-native";

import Button from "./Button";

storiesOf("Button", module).add("Button", () => (
  <Button>
    <Text>Button</Text>
  </Button>
));
