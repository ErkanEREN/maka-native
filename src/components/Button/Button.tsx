import React, { ReactNode } from "react";
import { View, TouchableHighlight } from "react-native";
import styled from "styled-components";

interface Props {
  onPress?: () => void;
  children: ReactNode;
}

export default function Button({ onPress, children }: Props) {
  return <TouchableHighlight onPress={onPress}>{children}</TouchableHighlight>;
}
