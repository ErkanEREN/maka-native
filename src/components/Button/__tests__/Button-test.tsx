import React from "react";
import { shallow } from "enzyme";
import Button from "../Button";
import { Text } from "react-native";
import renderer from "react-test-renderer";

describe("Button", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const children = <Text>Button Children</Text>;
      const tree = renderer
        .create(<Button onPress={() => {}}>{children}</Button>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
