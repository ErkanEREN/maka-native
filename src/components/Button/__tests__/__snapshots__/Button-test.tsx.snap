// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<div>
  <div>
    Button Text
  </div>
</div>
`;
